src/core/proj/qgscoordinatereferencesystem.cpp

```diff
@@ -2134,7 +2133,7 @@ void getOperationAndEllipsoidFromProjString( const QString &proj, QString &opera
     // and will result in oddities within other areas of QGIS (e.g. project ellipsoid won't be correctly
     // set for these CRSes). Better just hack around and make the constraint happy for now,
     // and hope that the definitions get corrected in future.
-    ellipsoid.clear();
+    ellipsoid = "";
```

Comment:
What's this for?

Reply:
to ensure the returned ellipsoid is not NULL but and empty string eventually.

Reply:
this reverst https://github.com/qgis/QGIS/commit/9e0bf6df993a8548be3e21e274ee2857728e658d#diff-0eb6084517fa5891245c56de95f3c07dc2424ffc25b732059acce2bdcfb6c66fL2461-R2480

Reply:
Does this void the not null constraint? In this case, is the not null constraint justified?

Reply:
I don't know. See the comment:
```
    // satisfy not null constraint on ellipsoid_acronym field
    // possibly we should drop the constraint, yet the crses with missing ellipsoid_acronym are malformed
    // and will result in oddities within other areas of QGIS (e.g. project ellipsoid won't be correctly
    // set for these CRSes). Better just hack around and make the constraint happy for now,
    // and hope that the definitions get corrected in future.
```

Reply:
Anyway, probably there aren't CRSs **without** ellipsoid/datum name in proj strings retrieved from PROJ database.

The relevant fix is to substitute
```c++
  if ( !ellipseMatch.hasMatch() )
  {
    ellipsoid = ellipseMatch.captured( 1 );
  }
```
with
```c++
  if ( ellipseMatch.hasMatch() )
  {
    ellipsoid = ellipseMatch.captured( 1 );
  }
```
to ensure that CRSs **with** ellipsoid/datum name in proj string will be properly imported.